contracts/src/PDPService.sol

```diff
@@ -131,33 +138,39 @@ contract PDPService {
 
         proofSetLeafCount[setId] = 0;
         proofSetOwner[setId] = address(0);
+        nextChallengeEpoch[setId] = 0;
     }
 
     // Struct for tracking root data
     struct RootData {
-        Cid root;
+        Cids.Cid root;
         uint256 rawSize;
     }
 
     function addRoots(uint256 setId, RootData[] calldata rootData) public returns (uint256) {
         require(proofSetLive(setId), "Proof set not live");
         require(rootData.length > 0, "Must add at least one root");
         require(proofSetOwner[setId] == msg.sender, "Only the owner can add roots");
+        bool needsChallengeEpoch = nextChallengeEpoch[setId] == 0;
         uint256 firstAdded = nextRootId[setId];
 
         for (uint256 i = 0; i < rootData.length; i++) {
             addOneRoot(setId, i, rootData[i].root, rootData[i].rawSize);
         }
+        // Initialise the first challenge epoch when the first data is added.
+        if (needsChallengeEpoch && proofSetLeafCount[setId] > 0) {
```

Comment:
I'd prefer dropping the second check.  Since addOneRoot only succeeds if size > 0 it is redundant.